src/Microsoft.AspNet.Server.Kestrel/Http/Frame.Generated.cs

```diff
@@ -0,0 +1,297 @@
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Server.Kestrel.Http 
+{
+    public partial class Frame
+    {
+        
+        private object _currentIHttpRequestFeature;
+        private object _currentIHttpResponseFeature;
+        private object _currentIHttpRequestIdentifierFeature;
+        private object _currentIHttpSendFileFeature;
+        private object _currentIServiceProvidersFeature;
+        private object _currentIHttpAuthenticationFeature;
+        private object _currentIHttpRequestLifetimeFeature;
+        private object _currentIQueryFeature;
+        private object _currentIFormFeature;
+        private object _currentIResponseCookiesFeature;
+        private object _currentIItemsFeature;
+        private object _currentIHttpConnectionFeature;
+        private object _currentITlsConnectionFeature;
+        private object _currentIHttpUpgradeFeature;
+        private object _currentIHttpWebSocketFeature;
+        private object _currentISessionFeature;
```

Comment:
It might also make a difference that the Frame object is allocated per-connection rather than per-request? So these objects fields aren't directly contributing to per-request memory pressure quite as aggressively.

The `List` vs `Dictionary` is a really interesting one. I do recall hearing that for small populations list scanning outperforms a hash table. Since the features added per request will always be a very limited set it might be better to use List instead of Dictionary all the time. Definitely worth measuring.

That said - we just declared all the well-known features used by our benchmarks as member fields... So it wouldn't actually make a difference to what we're measuring at this point. :\